test/test_multidevice_communications.cpp

```diff
@@ -0,0 +1,250 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#ifdef USE_DISTRIBUTED
+#include <gtest/gtest.h>
+
+#include <multidevice/communication.h>
+#include <multidevice/communicator.h>
+#include <test/utils.h>
+
+#include <iostream>
+
+namespace nvfuser {
+
+static constexpr DeviceIdxType root = 0;
+static constexpr int tensor_size = 1024;
+static constexpr int number_of_repetitions = 8;
+
+TEST_F(MultiDeviceTest, FusionMultiGPU_Communication_Gather_CUDA) {
```

Comment:
Our new scheme for naming tests does not have `Fusion` prefix and `_CUDA` suffix. These were legacies carried from PyTorch. New tests should use new naming scheme.